src/solvers.jl

```diff
@@ -195,3 +196,47 @@ function solve(solver::TruncatedSVD, A, y)
     return Dict{String, Any}("C" => solver.P \ θP)
 end
 
+
+@doc raw"""
+`struct ASP` : Active Set Pursuit sparse solver
+    solves the following optimization problem using the homotopy approach:
+
+    ```math 
+    \max_{y} \left( b^T y - \frac{1}{2} λ y^T y \right)
+    ```
+        subject to
+        
+    ```math
+        \|A^T y\|_{\infty} \leq 1.
+    ```
+
+    * Input
+    * `A` : `m`-by-`n` explicit matrix or linear operator.
+    * `b` : `m`-vector.
+    * `min_lambda` : Minimum value for `λ`. Is set to zero if not input is given.
+    * `loglevel` : Logging level.
+    * `itnMax` : Maximum number of iterations.
+    * `feaTol` : Feasibility tolerance.
+    * `actMax` : Maximum number of active constraints.
+
+    Constructor
+    ```julia
+    ACEfit.ASP(; P = I)
+    ``` 
+    where 
+    * `P` : right-preconditioner / tychonov operator
+"""
+struct ASP
+    P::Any
+end
+
+ASP(; P = I) = ASP(P)
+
+function solve(solver::ASP, A, y; kwargs...)
+    AP = A / solver.P
+    tracer = asp_homotopy(AP, y; loglevel=0, traceFlag=true, kwargs...)
+    xs = tracer[end][1]
```

Comment:
Let's also revisit this: I don't think that selecting the final iterate is normally desired. Maybe there could be a few options given by an additional parameters 
```julia 
ASP(; P = I, select = ( :byerror, 1.2 ), params... )
ASP(; P = I, select = ( :bysize, 1000 ), params... )
ASP(; P = I, select = ( :final, ), params... )
```

`(:byerror, 1.2)` would select the smallest active set fit that is within a factor 1.2 of the smallest fit error (not test error) achieved along the path. `(:bysize, 1000)` would select the smallest error fit that that less than 1000 active basis functions; while `(:final, )` would do exactly what you did. 

The only question is what the default should be. Maybe there should be no default and the user has to specify the mode?

Reply:
I guess :bysize here doesn't really make sense unless we just integrate it into SmartASP and bring in test/validation sets, because using homotopy, the residual on the training set monotonically decreases so the smallest error fit that has <= 1000 active basis functions happens at actMax=1000.

Reply:
other than this, I think I've made all of the changes requested! Please let me know if I should change anything else :)

Reply:
> monotonically decreases so the smallest error fit

ok, fair point. I need to think about it for a bit ...

Reply:
I think we should still add this option. I can envision that I want the solution with N parameters but have access to the model path up to 3*N parameters (or similar) for analysis ...